### DISCUSSION TITLE
knn ordering with <->

### DISCUSSION THREAD

**Discussion by berndhopp** [hibernate/hibernate-orm#4594] [Design Proposals]
k-nearest-neighbour ordering is implemented in postgis https://postgis.net/workshops/postgis-intro/knn.html 
would it make sense to introduce the knn order operator <-> as a custom jpql operator in hibernate spatial? on the one hand, it seems to be a useful feature, on the other hand it is not supported on mysql and mariadb AFAICS.

**Comment by gavinking** [comment]
This seems to me like a perfect case for just registering a function (which is _very_ easy to do). I would not introduce a new *operator* into HQL for this.

**Reply by beikov** [reply]
I agree, just use a function
